kythe/cxx/indexer/proto/proto_graph_builder.cc

```diff
@@ -211,4 +211,10 @@ void ProtoGraphBuilder::AddCodeFact(const VName& element,
   recorder_->AddMarkedSource(VNameRef(element), code);
 }
 
+void ProtoGraphBuilder::SetDeprecated(const VName& node_name) {
+  VLOG(1) << "Adding deprecation tag for " << StringifyNode(node_name);
+  recorder_->AddProperty(VNameRef(node_name), kythe::PropertyID::kTagDeprecated,
+                         "This field is deprecated.");
```

Comment:
I'd leave the fact value the empty string. If there was an explicit deprecation message to capture, that'd be useful but I don't think there is for proto.